### PR TITLE
FlowForge v1.4 release

### PR DESCRIPTION
**Pull request by ZJvandeWeg**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+**Automative Device Provisioning**
+
+FlowForge 1.4 now includes automatic device onboarding for fleets of devices. Simply download a FlowForge device provisioning credential to a new device or fleet of devices. On device restart, the FlowForge agent and the Node-RED snapshot will automatically be provisioned to the device. [Issue #1212](https://github.com/flowforge/flowforge/issues/1212)
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/B7XK3TUklUU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+**Support for Staged Development**
+
+A new feature of 1.4 is the ability to setup staged deployments. This makes it possible to simply move a project between a Development > Test > Production for your Node-RED application delivery.  [Issue #1580](https://github.com/flowforge/flowforge/issues/1580)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:31]:
```suggestion
A new feature of 1.4 is the ability to setup staged deployments. This makes it possible to simply move a project between a Development > Test > Production for your Node-RED application delivery. [Issue #1580](https://github.com/flowforge/flowforge/issues/1580)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/JRk-Cf7eNIo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:35]:
```suggestion
```

**Comment by iskerrett** [comment]
The blog post is ready for review. @robmarcer still needs to add in the videos for the two features.

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+- Show a preview of Team Library entries on click. [#1657](https://github.com/flowforge/flowforge/issues/1657)
+
+- Add a synchronous mode for the FlowForge context store. [#17](https://github.com/flowforge/flowforge-nr-persistent-context/issues/17)
+
+- Add a Last Seen status for devices connecting to FlowForge. [#1599](https://github.com/flowforge/flowforge/issues/1599)
+
+- Added a check to ensure the team slug is unique. [#1609](https://github.com/flowforge/flowforge/issues/1609)
+
+- Optionally set snapshot as target at creation [#1527](https://github.com/flowforge/flowforge/issues/1527)
+
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:50]: Agents will keep checking the forge endpoint, agents can now be provisioned without internet connection for example

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+We're pleased to announce version 1.4 is now available!  Keep reading for the details of what's in this release or you can watch our 1 minute roundup video of the new release above.
+
+To make it easy for everyone to experience FlowForge, we are introducing a new free 30-day trial. You can now experience the power of using FlowForge to quickly deliver Node-RED applications in a reliable, repeatable, collaborative, and secure manner. To get your free trial simply [sign up for FlowForge Cloud;](https://app.flowforge.com/account/create) no credit card is required!.
+
+## New User Features
+
+
+**Automatic Device Provisioning**
+
+FlowForge 1.4 now includes automatic device onboarding for fleets of devices. Simply download a FlowForge device provisioning credential to a new device or fleet of devices. On device restart, the FlowForge agent and the Node-RED snapshot will automatically be provisioned to the device. [Issue #1212](https://github.com/flowforge/flowforge/issues/1212)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:24]:
```suggestion
Most prominently, FlowForge 1.4 features automatic device onboarding for fleets. Simply download a FlowForge device provisioning credential to allow quick roll-out to a whole fleet, without the need to have device specific configuration. When the agent starts, the FlowForge agent and the Node-RED snapshot will automatically be provisioned to the device and start operations. [Issue #1212](https://github.com/flowforge/flowforge/issues/1212)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/B7XK3TUklUU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+**Support for Staged Development**
+
+A new feature of 1.4 is the ability to setup staged deployments. This makes it possible to simply move a project between a Development > Test > Production for your Node-RED application delivery. [Issue #1580](https://github.com/flowforge/flowforge/issues/1580)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:31]:
```suggestion
Staged deployments improve reliability of flows developed. This is why FlowForge now allows DevOps-style pipelines to be employed. This makes it possible to simply move a project between a Development > Test > Production for your Node-RED application delivery. [Issue #1580](https://github.com/flowforge/flowforge/issues/1580)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/JRk-Cf7eNIo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+
+## Improvements
+
+- It is now possible to change the project type. This includes changing the version of Node-RED being used for the project and the small/medium/large.  [#595](https://github.com/flowforge/flowforge/issues/595)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:39]:
```suggestion
- It's now much easier to change the resources available to your Node-RED instance. With a few clicks a resource intensive workload can be processed faster by changing between small, medium, or large instance types. [#595](https://github.com/flowforge/flowforge/issues/595)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+- Show a preview of Team Library entries on click. [#1657](https://github.com/flowforge/flowforge/issues/1657)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:41]:
```suggestion
- Last release allows users to capture flows in a shared library to reuse in another flow, now it's possible to preview the stored flows. [#1657](https://github.com/flowforge/flowforge/issues/1657)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+- Add a synchronous mode for the FlowForge context store. [#17](https://github.com/flowforge/flowforge-nr-persistent-context/issues/17)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:43]:
```suggestion
- Add a synchronous mode for the FlowForge persisted context store, next to the asynchronous mode already available. [#17](https://github.com/flowforge/flowforge-nr-persistent-context/issues/17)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+- Add a Last Seen status for devices connecting to FlowForge. [#1599](https://github.com/flowforge/flowforge/issues/1599)
+
+- Added a check to ensure the team slug is unique. [#1609](https://github.com/flowforge/flowforge/issues/1609)
+
+- Optionally set snapshot as target at creation [#1527](https://github.com/flowforge/flowforge/issues/1527)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:49]:
```suggestion
- Optionally set snapshot as target at creation, to quickly roll out changes to remote deployments [#1527](https://github.com/flowforge/flowforge/issues/1527)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md (hunk 3 above)
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:50]:
```suggestion
- Agents are more rugged when starting up if they're unable to connect to FlowForge, and will retry to connect.

```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
+---
+title: FlowForge 1.4 is now available, FEATURE and much more
+subtitle: Our second release of 2023 with some great new features to try out.
+description: FlowForge v1.4 is now available, FEATURE
+date: 2023-02-16 18:00:00.0
+authors: ["ian-skerrett"]
+video: 
+image: /blog/2023/01/images/flowforge-130-hero.png
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:8]: Is this correct?

```suggestion
image: /blog/2023/01/images/flowforge-130-hero.png
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
+---
+title: FlowForge 1.4 is now available, FEATURE and much more
+subtitle: Our second release of 2023 with some great new features to try out.
+description: FlowForge v1.4 is now available, FEATURE
+date: 2023-02-16 18:00:00.0
+authors: ["ian-skerrett"]
+video:
```
Comment (robmarcer) [inline src/blog/2023/02/flowforge-1-4-0-released.md:7]: [vbg4zTmUYjQ](https://youtu.be/vbg4zTmUYjQ)

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+## New User Features
+
+
+**Automatic Device Provisioning**
+
+FlowForge 1.4 now includes automatic device onboarding for fleets of devices. Simply download a FlowForge device provisioning credential to a new device or fleet of devices. On device restart, the FlowForge agent and the Node-RED snapshot will automatically be provisioned to the device. [Issue #1212](https://github.com/flowforge/flowforge/issues/1212)
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/B7XK3TUklUU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (robmarcer) [inline src/blog/2023/02/flowforge-1-4-0-released.md:27]: https://youtu.be/XTVw4O4-Crg

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+**Support for Staged Development**
+
+A new feature of 1.4 is the ability to setup staged deployments. This makes it possible to simply move a project between a Development > Test > Production for your Node-RED application delivery. [Issue #1580](https://github.com/flowforge/flowforge/issues/1580)
+
+
+<div><iframe width="560" height="315" src="https://www.youtube.com/embed/JRk-Cf7eNIo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (robmarcer) [inline src/blog/2023/02/flowforge-1-4-0-released.md:34]: https://youtu.be/6QOmotlrwWw

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+## Bug Fixes
+
+We've fixed the following bugs in this release.
+- Deleting your only team, doesn't exit from the team UI. [#1630](https://github.com/flowforge/flowforge/issues/1630)
+- Async Team Slug Check  [#1609](https://github.com/flowforge/flowforge/issues/1609)
+- Improve communication of Device Last Seen and Status [#1599](https://github.com/flowforge/flowforge/issues/1599)
+
+
+## Contributors
+
```
Comment (robmarcer) [inline src/blog/2023/02/flowforge-1-4-0-released.md:66]: We'd like the thank the following for their contributions to this release:

[UlisesGascon](https://github.com/UlisesGascon) for their work on [#74](https://github.com/flowforge/installer/pull/74)

**Review by robmarcer — CHANGES_REQUESTED** [review]

File: src/blog/2023/02/flowforge-1-4-0-released.md (hunk 10 above)
Comment (iskerrett) [reply src/blog/2023/02/flowforge-1-4-0-released.md]: I have added the video and image now.

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+## Getting help
+
+Please check FlowForge's [documentation](https://flowforge.com/docs/) as the answers to many questions are covered there.
+
+If you hit any problems with the platform please raise an [issue on GitHub](https://github.com/flowforge/flowforge/issues).
+That's also a great place to send us any feedback or feature requests.
+
+You can also get help on [the Node-RED forums](https://discourse.nodered.org/)
+
+AS well as in the [forum within our Github project](https://github.com/flowforge/flowforge/discussions)
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:96]:
```suggestion
As well as in the [forum within our Github project](https://github.com/flowforge/flowforge/discussions)
```

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
+---
+title: FlowForge v1.4 with device provisioning in bulk and staged development process
+subtitle: Our second release of 2023 with some great new features to try out.
+description: Deploy Node-RED to many devices quickly, and allow a staged development process with the latest release of FlowForge v1.4.
+date: 2023-02-16 18:00:00.0
```
Comment (ZJvandeWeg) [inline src/blog/2023/02/flowforge-1-4-0-released.md:5]: If we merge before 1800 UTC, note it might not appear on our blog right away

**Review by iskerrett — APPROVED** [review]

**Review by iskerrett — APPROVED** [review]

File: src/blog/2023/02/flowforge-1-4-0-released.md (hunk 16 above)
Comment (iskerrett) [reply src/blog/2023/02/flowforge-1-4-0-released.md]: Fixed

File: src/blog/2023/02/flowforge-1-4-0-released.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
+---
+title: FlowForge v1.4 with device provisioning in bulk and staged development process
+subtitle: Our second release of 2023 with some great new features to try out.
+description: Deploy Node-RED to many devices quickly, and allow a staged development process with the latest release of FlowForge v1.4.
+date: 2023-02-16 14:00:00.0
+authors: ["ian-skerrett"]
+video: vbg4zTmUYjQ
+image: /blog/2023/0102/images/ff-r14-image.png
```
Comment (joepavitt) [inline src/blog/2023/02/flowforge-1-4-0-released.md:8]: @iskerrett looks like the file path is wring, should just be 02/ according to where the image was added

**Pull request merged**